test/integration/webhook_test.go

```diff
@@ -23,534 +17,13 @@ import (
-			Name: "brokenfence",
-		},
-		StringData: map[string]string{
-			"kongCredType": "basic-auth",
-			"username":     "brokenfence",
-			"password":     "testpass",
-		},
-	}, metav1.CreateOptions{})
-	require.NoError(t, err)
-
-	t.Log("verifying that valid credentials assigned to a consumer pass validation")
-	validConsumerLinkedToValidCredentials := &kongv1.KongConsumer{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: uuid.NewString(),
-			Annotations: map[string]string{
-				annotations.IngressClassKey: consts.IngressClass,
-			},
-		},
-		Username: "brokenfence",
-		CustomID: uuid.NewString(),
-		Credentials: []string{
-			"brokenfence",
-		},
-	}
-	validConsumerLinkedToValidCredentials, err = kongClient.ConfigurationV1().KongConsumers(ns.Name).Create(ctx, validConsumerLinkedToValidCredentials, metav1.CreateOptions{})
-	require.NoError(t, err)
-	defer func() {
-		if err := kongClient.ConfigurationV1().KongConsumers(ns.Name).Delete(ctx, validConsumerLinkedToValidCredentials.Name, metav1.DeleteOptions{}); err != nil {
-			if !apierrors.IsNotFound(err) {
-				assert.NoError(t, err)
-			}
-		}
-	}()
-
-	t.Log("verifying that the valid credentials which include a unique-constrained key can be updated in place")
-	validCredential.Data["value"] = []byte("newpassword")
-	validCredential, err = env.Cluster().Client().CoreV1().Secrets(ns.Name).Update(ctx, validCredential, metav1.UpdateOptions{})
-	require.NoError(t, err)
-
-	t.Log("verifying that validation fails if the now referenced and valid credential gets updated to become invalid")
-	validCredential.Data["kongCredType"] = []byte("invalid-auth")
-	_, err = env.Cluster().Client().CoreV1().Secrets(ns.Name).Update(ctx, validCredential, metav1.UpdateOptions{})
-	require.Error(t, err)
-	require.Contains(t, err.Error(), "invalid credential type")
-
-	t.Log("verifying that if the referent consumer goes away the validation fails for updates that make the credential invalid")
-	require.NoError(t, kongClient.ConfigurationV1().KongConsumers(ns.Name).Delete(ctx, validConsumerLinkedToValidCredentials.Name, metav1.DeleteOptions{}))
-	_, err = env.Cluster().Client().CoreV1().Secrets(ns.Name).Update(ctx, validCredential, metav1.UpdateOptions{})
-	require.ErrorContains(t, err, "invalid credential type")
-
-	t.Log("verifying that a JWT credential which has keys with missing values fails validation")
-	invalidJWTName := uuid.NewString()
-	invalidJWT := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: invalidJWTName,
-		},
-		StringData: map[string]string{
-			"kongCredType":   "jwt",
-			"algorithm":      "RS256",
-			"key":            "",
-			"rsa_public_key": "",
-			"secret":         "",
-		},
-	}
-	_, err = env.Cluster().Client().CoreV1().Secrets(ns.Name).Create(ctx, invalidJWT, metav1.CreateOptions{})
-	require.ErrorContains(t, err, "some fields were invalid due to missing data: rsa_public_key, key, secret")
-}
-
 func ensureWebhookService(ctx context.Context, t *testing.T, name string) {
```

Comment:
There are no dedicated webhook test cases in integration tests, how about moving the helper functions in `helper_tests` or mark them as `t.Helper()`?

Reply:
Changed.